trait2d/analysis/__init__.py

```diff
@@ -1049,26 +1054,35 @@ def normalized(self, normalize_t = True, normalize_xy = True):
         return NormalizedTrack(x, y, t, xmin, ymin, tmin, id=self._id)
 
     def calculate_msd(self):
-        """Calculates the mean squared displacement of the track. Result is stored in the Track object.
+        
+        """Calculates the track's mean squared displacement (msd) and stores it in 'track' object. Also deletes temporary values and colelcts them.
+           Furthermore, calculates the standard deviation per point of the MSD array (msd_error) and the standard error of the mean (SEM) [legacy version].
         """
+        
+        import gc
+                
+        N = len(self._x)
+        col_Array  = np.zeros(N-3)
+        Err_col_Array = np.zeros(N-3)
+
+        data_tmp = np.column_stack((self._x, self._y))
+    
+        for i in tqdm.tqdm(range(1,N-2)):
+            calc_tmp = np.sum(np.abs((data_tmp[1+i:N,:] - data_tmp[1:N - i,:]) ** 2), axis=1)
+            col_Array[i-1] = np.mean(calc_tmp)
+            Err_col_Array[i-1] = np.std(calc_tmp)
+            
+            del calc_tmp
```

Comment:
Deleting `calc_tmp` is not necessary as it goes out of scope anyways. (See next comment as well.)

Reply:
I just thought of it being cleaner if you reset after each run.

Reply:
I mean, `del` definitely won't break anything here. It won't do anything at all, in fact :D